include/deal.II/lac/la_parallel_block_vector.h

```diff
@@ -469,6 +469,39 @@ namespace LinearAlgebra
                                      const BlockVector<Number> &V,
                                      const bool symmetric = false) const;
 
+      /**
+       * Calculate the scalar product between each block of this vector and @p V
+       * using a metric tensor @p matrix. This function
+       * computes the result of $ \sum_{ij} A^{ij} U_i \cdot V_j$ where $U_i$
+       * and $V_j$ indicate the $i$th block (not element) of $U$ and the
+       * $j$th block of $V$, respectively. If @p symmetric is
+       * <code>true</code>, it is assumed that $U_i \cdot V_j$ and $A^{ij}$ are
+       * symmetric matrices and almost half of the scalar products can be avoided.
+       *
+       * Obviously, this function can only be used if all blocks of both vectors
+       * are of the same size.
+       *
+       * @note Internally, a single global reduction will be called to
+       * accumulate the scalar product between locally owned degrees of freedom.
+       */
+      template <typename FullMatrixType>
+      Number multivector_inner_product_with_metric(const FullMatrixType &matrix,
+                                                   const BlockVector<Number> &V,
+                                                   const bool symmetric = false) const;
+
+      /**
+       * Set each block of this vector as follows:
+       * $U^i = \sum_{j} A^{ij} V_j$ where $U^i$
+       * and $V_j$ indicate the $i$th block (not element) of $U$ and the
+       * $j$th block of $V$, respectively.
+       *
+       * Obviously, this function can only be used if all blocks of both vectors
+       * are of the same size.
+       */
+      template <typename FullMatrixType>
+      void equ(const FullMatrixType &matrix,
+               const BlockVector<Number> &V);
```

Comment:
Does anybody have a better name for this? I have the feeling that the meaning of `equ` in the scalar context is not really the appropriate extension. Essentially, this operation is a matrix-matrix product with `V` the left matrix (long and skinny) and `matrix^T` the right matrix. Any opinions?

Reply:
It’s actually V= M*U where V and U are matrices with columns in block vectors. So apart from ‘mmult’ I don’t think there is another naming option.

Reply:
(I meant V as this object, of course)

Reply:
Should be V=U*M^T , my bad

Reply:
maybe `mmult(const Vec&, const FullMatrix&)` so that the order of arguments imply `U=V*M` (i will also change the indices in M)

Reply:
Indeed, I think `mmult` is a good name for this with the order of arguments as you propose in the last post